### PR TITLE
Create database schema, populate it and add basic endpoints

### PR DESCRIPTION
**Pull request by vdrelciuc**
This PR contains:
- SQL schema and mock data (a few basic rooms with questions, answers and participants)
- Common interfaces that can be used both by the client and the server (they define the data structure)
- Database debugging endpoints to list all elements of a table and to reset the mock data
- 3 basic endpoints: get all questions given an event key, get all choices given a question id, post an answer given question id and participant id
- Updated README.MD to include instructions on how to set up the database locally for testing purposes
Warning: I'll need to provide you a .env file containing all the API keys before you'll be able to access the database

### CODE REVIEW

File: server/app/services/database.service.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    async getAllChoicesByQuestion(questionId: string): Promise<QueryResult> {
+        return this.pool.query(`
+        SELECT * FROM ${this.SCHEMA_NAME}.Choice c
+        WHERE c.question_id = '${questionId}';`);
+    }
+
+    async createAnswer(questionId: string, participantId: string, answerLabel: string) {
+        return this.pool.query(`
+        INSERT INTO ${this.SCHEMA_NAME}.AnswerEntry (question_id, participant_id, answer_label) 
+        VALUES (${questionId},${participantId},'${answerLabel}');`);
```
Comment (nitinms-sps-program) [inline server/app/services/database.service.ts:60]: What happens if answerLabel has a quote in it?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,24 @@
 # Online Quiz Platform
 A tool for educators to engage their students throughout class.
 
+## Basic Database Setup (Dev Environment)
+### Method #1 - IP Whitelist (faster, less optimal)
+1. Go to the [Google Cloud Console](https://console.cloud.google.com/)
+2. After selecting the right project, navigate to SQL
+3. Select the right instance
+4. Under Connections -> Authorized networks, add your IP Address
+5. Don't forget to upload your .env file to /server
```
[the remark this answers is not in the document]
Comment (vdrelciuc) [reply README.md:10]: Copying the .env file inside the local checkout, yes :)

File: server/app/controllers/choice.controller.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                .then((result: QueryResult) => {
+                    const choices: Choice[] = result.rows.map((choice: any) => ({
+                        choiceId: choice.choice_id,
+                        choiceLabel: choice.choice_label,
+                        questionId: choice.question_id
+                    }));
+                    res.status(HttpStatus.OK).send(choices);
+                })
+                .catch((e: Error) => {
+                    res.status(HttpStatus.BAD_REQUEST).send(e.message);
```
[the remark this answers is not in the document]
Comment (vdrelciuc) [reply server/app/controllers/choice.controller.ts:28]: It won't, we'll deal with the front-end error management using the HTTP status codes (e.g. telling the user that one of his entries is incorrect). The e.message is only for debugging purposes. Would you like me to take it off completely?

File: common/interfaces/quiz.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export interface Quiz {
+    quizId: number;
+    maxDuration: number;
+    roomId: number;
```
Comment (amulyaparmar) [inline common/interfaces/quiz.ts:4]: What happens if the quiz does not have a Room ID?

Comment (vdrelciuc) [reply common/interfaces/quiz.ts:4]: Depends on the endpoint implementation. I haven't had the chance to code an endpoint that needed a quiz ID so far, but if I were to write one, it'd send a 400 BAD REQUEST answer to the client if it was missing arguments

File: server/app/services/database.service.ts (hunk 1 above)
Comment (vdrelciuc) [reply server/app/services/database.service.ts:60]: I'm working on a way to sanitize user input on the server side. I'll update my PR as soon as I get it working

Comment (vdrelciuc) [reply server/app/services/database.service.ts]: Fixed in commit 6584c0eed

**Comment by vdrelciuc** [comment]
I've added unit tests for both the controllers and the database service. I also fixed the SQL injection vulnerability by using parameterized queries. This also takes care of input sanitization (such as single quotes and double quotes).

**Pull request merged**